sdcm/tester.py

```diff
@@ -248,40 +249,47 @@ def setUp(self):
+        try:
+            self.init_resources()
 
-        self.init_resources()
-
-        self.init_nodes(db_cluster=self.db_cluster)
-
-        # cs_db_cluster is created in case MIXED_CLUSTER. For example, gemini test
-        if self.cs_db_cluster:
-            self.init_nodes(db_cluster=self.cs_db_cluster)
-
-        if self.create_stats:
-            self.create_test_stats()
-            # sync test_start_time with ES
-            self.start_time = self.get_test_start_time()
+            self.init_nodes(db_cluster=self.db_cluster)
 
-        self.set_system_auth_rf()
+            # cs_db_cluster is created in case MIXED_CLUSTER. For example, gemini test
+            if self.cs_db_cluster:
+                self.init_nodes(db_cluster=self.cs_db_cluster)
 
-        db_node_address = self.db_cluster.nodes[0].ip_address
-        self.loaders.wait_for_init(db_node_address=db_node_address)
+            if self.create_stats:
+                self.create_test_stats()
+                # sync test_start_time with ES
+                self.start_time = self.get_test_start_time()
 
-        if self.params.get("use_mgmt", default=None):
-            mgmt_auth_token = str(uuid4())
-            for node in self.db_cluster.nodes:
-                node.install_manager_agent(mgmt_auth_token, self.params.get("scylla_mgmt_repo"))
-            self.monitors.wait_for_init(auth_token=mgmt_auth_token)
-        else:
-            self.monitors.wait_for_init()
+            self.set_system_auth_rf()
 
-        # cancel reuse cluster - for new nodes added during the test
-        cluster.Setup.reuse_cluster(False)
-        if self.monitors.nodes:
-            self.prometheus_db = PrometheusDBStats(host=self.monitors.nodes[0].public_ip_address)
-        self.start_time = time.time()
+            db_node_address = self.db_cluster.nodes[0].ip_address
+            self.loaders.wait_for_init(db_node_address=db_node_address)
 
-        self.db_cluster.validate_seeds_on_all_nodes()
+            if self.params.get("use_mgmt", default=None):
+                mgmt_auth_token = str(uuid4())
+                for node in self.db_cluster.nodes:
+                    node.install_manager_agent(mgmt_auth_token, self.params.get("scylla_mgmt_repo"))
+                self.monitors.wait_for_init(auth_token=mgmt_auth_token)
+            else:
+                self.monitors.wait_for_init()
+
+            # cancel reuse cluster - for new nodes added during the test
+            cluster.Setup.reuse_cluster(False)
+            if self.monitors.nodes:
+                self.prometheus_db = PrometheusDBStats(host=self.monitors.nodes[0].public_ip_address)
+            self.start_time = time.time()
+
+            self.db_cluster.validate_seeds_on_all_nodes()
+        except Exception as exc:
+            TestFrameworkEvent(
+                source=self.__class__.__name__,
+                source_method='SetUp',
+                exception=exc
+            ).publish()
```

Comment:
any reason you didn't put the `.publish()` in `__init__` of `TestFrameworkEvent` ?

Reply:
In order to be able to create event and not publish it.
Could be handfull for testing, deduplication or any other task of such kind.

Reply:
o.k., I less like it.
cause from my POV it's would cause someone down the road to forget doing the `.publish()`, the common case is that we want to publish immediately, event in testing (which this event doesn't have...)